src/backends/ilp-quoter/index.js

```diff
@@ -85,28 +75,40 @@ class ILPQuoter {
                 currencyPair[0] + '/' + currencyPair[1] + '/' + amount +
                 '/' + type
     const result = yield request({ uri, json: true })
-    const fixedAmount = type === 'source' ? result.body.source_amount
-                                          : result.body.destination_amount
+    if (result.statusCode >= 400) {
+      log.error('Error getting quote: ', JSON.stringify(result.body))
+      throw new ServerError('Unable to get quote from backend.')
+    }
+    const source_amount = result.body.source_amount
+    const destination_amount = result.body.destination_amount
+    if (!source_amount || !destination_amount) {
+      log.error('Quoter did not return amounts: ', JSON.stringify(result.body))
+      throw new ServerError('Quoter returned invalid quote amounts.')
+    }
+    const fixedAmount = type === 'source' ? source_amount : destination_amount
     if (fixedAmount !== amount.toString()) {
       if (!(new BigNumber(fixedAmount).equals(new BigNumber(amount)))) {
         throw new UnacceptableQuoterAmountError('Backend returned an invalid ' +
                   type + ' amount: ' + fixedAmount + ' (expected: ' + amount + ')')
       }
     }
+
+    if (isNaN(source_amount) || Number(source_amount) === 0 || Number(source_amount) === Number.POSITIVE_INFINITY) {
```

Comment:
Maybe pull some of this validation into a `validate` function. Also, any reason to not use Bignumber here?